### PR TITLE
Add page explaining provider trust tiers

### PR DESCRIPTION
**Pull request by paultyng**

### CODE REVIEW

File: website/docs/registry/providers/tiers.html.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+description: |-
+  Published Provider tiers in the Terraform Registry
+---
+
+# Provider Tiers
+
+There are three tiers of providers in the Terraform Registry:
+
+* **Official Providers** - are built, signed, and supported by HashiCorp. Official Providers can typically used without providing
+  provider source information in your Terraform configuration.
```
Comment (justincampbell) [inline website/docs/registry/providers/tiers.html.md:14]: I believe this is going to be a requirement or warning in 0.13, and that users should use the upgrade tool to specify the explicit source, @hashicorp/terraform-core can confirm though.

File: website/docs/registry/providers/tiers.html.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+* **Partner Providers** - are built, signed, and supported by a third party. HashiCorp has verified the ownership of the private
+  key and we provide a chain of trust to the CLI to verify this programatically. To use Partner Providers in your Terraform
+  configuration, you need to specify the namespace and name to download from the registry.
```
Comment (mildwonkey) [inline website/docs/registry/providers/tiers.html.md:17]: It's more accurate (but not necessarily clearer) to say that provider source is required for partner providers. Only the namespace and provider name are required, but that's specifically in the `"source"` attribute under `terraform.required_providers`.  (this isn't really a suggestion and I won't think twice if you don't do anything with this information)

File: website/docs/registry/providers/tiers.html.md (hunk 1 above)
Comment (mildwonkey) [reply website/docs/registry/providers/tiers.html.md:14]: Not for official providers - if it's in the `hashicorp` namespace, source is not required.

File: website/docs/registry/providers/tiers.html.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+sidebar_current: "docs-registry-provider-tiers
+description: |-
+  Published Provider tiers in the Terraform Registry
+---
+
+# Provider Tiers
+
+There are three tiers of providers in the Terraform Registry:
+
+* **Official Providers** - are built, signed, and supported by HashiCorp. Official Providers can typically used without providing
```
Comment (alisdair) [inline website/docs/registry/providers/tiers.html.md:13]:
```suggestion
* **Official Providers** - are built, signed, and supported by HashiCorp. Official Providers can typically be used without providing
```

File: website/docs/registry/providers/tiers.html.md (hunk 2 above)
Comment (paultyng) [reply website/docs/registry/providers/tiers.html.md:17]:
```suggestion
  configuration, you need to specify the provider source, typically this is the namespace and name to download from the registry.
```

**Comment by paultyng** [comment]
Thanks everyone, incorporated the suggestions.

**Comment by justincampbell** [comment]
@paultyng did you want to add this to the nav too?

**Comment by paultyng** [comment]
Good call, will put up a new PR

**Pull request merged**